### PR TITLE
Chatbot workshop

### LINKED ISSUE #10: Build conversational AI workshop notebook
The current Lex bot integration on the Help view (accessed from "?" icon in top nav) will load the chat bot widget if there is a Lex bot configured on the backend. Otherwise, a message is displayed indicating that it's not fully configured. 

Building out the chatbot experience would make for a useful workshop. For this issue, a workshop notebook should be created that walks the user/developer through building a chatbot in Lex that provides retail-specific intents. Once the bot is created on the backend via the workshop, the frontend is already setup to automatically load and display the bot widget.

### PR DESCRIPTION
**Pull request by james-jory**
*Issue #, if available:*

Closes #10 

*Description of changes:*

I needed a conversational/chatbot workshop to round some features for an upcoming blog post.

- Required working around limitations in the Amplify chatbot component where session attributes could not be sent with PostText/PostContent calls (i.e. to pass the user ID) and the component does not support response cards. Linking the entityId sent by the chatbot to a store user required tracking the entityId with the user in the Users service. Unfortunate this was necessary but a workaround was needed. This was the cleanest approach that I could get working.
- A new Lambda function was added that is called by Lex to retrieve recommendations.
- A new workshop notebook was created to walk reader through setting up the chatbot and wiring up the Lambda function.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/aws-lambda/bot-intent-recommendations/bundle.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (BastLeblanc) [inline src/aws-lambda/bot-intent-recommendations/bundle.sh:1]: general comment on the bundle.sh and stage.sh.
It's the same for every Lambda function, maybe some opportunities to factor some code?

File: .gitignore [hunk 2]
```diff
@@ -16,3 +16,4 @@ demo.md
 generators/*.gz
 /csvs/
 .unotes/*
+!workshop/5-Conversational/RetailDemoStore_Lex.zip
```
Comment (BastLeblanc) [inline .gitignore:19]: why this zip in the gitignore? need it for import in Lex console, right?

File: src/aws-lambda/bot-intent-recommendations/bundle.sh (hunk 1 above)
Comment (james-jory) [reply src/aws-lambda/bot-intent-recommendations/bundle.sh:1]: Yes, there is a growing need to simplify how the growing number of lambdas are deployed. SAM is probably the answer.

File: .gitignore (hunk 2 above)
Comment (james-jory) [reply .gitignore:19]: Yes, the zip file is needed to import the chatbot config in Lex. It's very small. Just contains a JSON file.

**Pull request merged**